build/HelpProvider/Build.HelpProvider.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Markdig;
+using Markdig.Extensions.CustomContainers;
+using Markdig.Helpers;
+using Markdig.Syntax;
+using Markdig.Syntax.Inlines;
+using Nuke.Common;
+
+partial class Build : NukeBuild
+{
+    [Parameter("Shows help for the target", Name = "?")] public static bool HelpParameter = false;
+    protected override void OnTargetStart(string target)
+    {
+        if (HelpParameter)
+        {
+            Logger.Info(GetHelpForTarget(target));
+            Environment.Exit(0);
+        }
+        else base.OnTargetStart(target);
+    }
+
+    private string GetHelpForTarget(string target)
+    {
+        var pipeline = new MarkdownPipelineBuilder().UseCustomContainers().Build();
+        var vcbuildRoot = AppDomain.CurrentDomain.BaseDirectory;
+        var docsPath = Path.Combine(vcbuildRoot, "Docs", "Targets.md");
+        var md = Markdown.Parse(File.ReadAllText(docsPath), pipeline);
+        var containers = md.Descendants<CustomContainer>();
+        var container = containers.FirstOrDefault(c =>
+        {
+            var heading = c.Descendants<HeadingBlock>().FirstOrDefault();
+            if (heading == null)
+                return false;
+
+            return target == GetTextContent(heading);
+        });
+        if (container == null)
+        {
+            Logger.Error($"Help is not found for the target {target}");
+            return string.Empty;
+        }
+        var descriptionBlock = container.Descendants<ParagraphBlock>().FirstOrDefault();
+        var exampleBlock = container.Descendants<FencedCodeBlock>().FirstOrDefault();
+        var description = descriptionBlock != null ? GetTextContent(descriptionBlock) : string.Empty;
```

Comment:
GetTextContent already implement nullable check. No reason for checking it here.
```suggestion
        var description = GetTextContent(descriptionBlock);
```